Userland/Shell/Parser.cpp

```diff
@@ -1982,8 +1982,10 @@ RefPtr<AST::Node> Parser::parse_heredoc_initiation_record()
 
 bool Parser::parse_heredoc_entries()
 {
+    auto heredocs = move(m_heredoc_initiations);
+    m_heredoc_initiations.clear();
```

Comment:
Move-constructing the `heredocs` vector from another vector will clear out the source vector. No need to call `clear()` again :)

Reply:
tbh that's for the reader (i.e. future me) - "no, this function won't leave the vector in a weird state, don't go looking"